lua/resty/moesif/moesif_ser.lua

```diff
@@ -165,55 +165,72 @@ function _M.prepare_message(config)
     user_id_entity = ngx.var.remote_user
   elseif ngx.var.application_id ~= nil then
     user_id_entity = ngx.var.application_id
-  elseif request_headers[string.lower(config:get("authorization_header_name"))] ~= nil and string.lower(config:get("authorization_user_id_field")) ~= nil then
-
+  elseif string.lower(config:get("authorization_header_name")) ~= nil and string.lower(config:get("authorization_user_id_field")) ~= nil then
+    -- Split authorization header name by comma
+    local auth_header_names = split(string.lower(config:get("authorization_header_name")), ",")    
+    local token = nil
     -- Fetch the token and field from the config
-    local token = request_headers[string.lower(config:get("authorization_header_name"))]
+    for _, name in pairs(auth_header_names) do
+      local auth_name = name:gsub("%s+", "")
+      if request_headers[auth_name] ~= nil then 
+        if type(request_headers[auth_name]) == "table" and (request_headers[auth_name][0] ~= nil or request_headers[auth_name][1] ~= nil) then 
```

Comment:
@keyur9  just got to look at this.  Why are we checking that actual header value is table?

Reply:
It could be case when same header is being sent twice in the request, for example - when `X-Auth` is sent twice, lua will convert it into table `X-Auth` = {[0] = "yy", [1] = "xx"} in the request headers. It will throw errors when we try to match `string.match` token because it was expecting string and got table. So in order to avoid errors, we check if the value is of type table, we get the first element from the table and if it is of type string, we use as it is.

Reply:
oh k. is it openresty specific or do we need the same fix for kong also then?

Reply:
Not sure, will check Kong and make necessary changes, if needed.

Reply:
created pivotal for porting this to kong. when doing that, you can check